datasets/librispeech_asr/librispeech_asr.py

```diff
@@ -0,0 +1,153 @@
+
+Note that in order to limit the required storage for preparing this dataset, the audio
+is stored in the .flac format and is not converted to a float32 array. To convert, the audio
+file to a float32 array, please make use of the `.map()` function as follows:
+
+
+```python
+import soundfile as sf
+
+def map_to_array(batch):
+    speech_array, _ = sf.read(batch["file"])
+    batch["speech"] = speech_array
+    return batch
+
+dataset = dataset.map(map_to_array, remove_columns=["file"])
+"""
+
+_URL = "http://www.openslr.org/12"
+_DL_URL = "http://www.openslr.org/resources/12/"
+
+_DL_URLS = {
+    "clean": {
+        "dev": _DL_URL + "dev-clean.tar.gz",
+        "test": _DL_URL + "test-clean.tar.gz",
+        "train.100": _DL_URL + "train-clean-100.tar.gz",
+        "train.360": _DL_URL + "train-clean-360.tar.gz",
+    },
+    "other": {
+        "test": _DL_URL + "test-other.tar.gz",
+        "dev": _DL_URL + "dev-other.tar.gz",
+        "train.500": _DL_URL + "train-other-500.tar.gz",
+    },
+}
+
+
+class LibrispeechASRConfig(datasets.BuilderConfig):
+    """BuilderConfig for LibriSpeechASR."""
+
+    def __init__(self, **kwargs):
+        """
+        Args:
+          data_dir: `string`, the path to the folder containing the files in the
+            downloaded .tar
+          citation: `string`, citation for the data set
+          url: `string`, url for information about the data set
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(LibrispeechASRConfig, self).__init__(version=datasets.Version("2.1.0", ""), **kwargs)
+
+
+class LibrispeechASR(datasets.GeneratorBasedBuilder):
+    """Librispeech dataset."""
+
+    BUILDER_CONFIGS = [
+        LibrispeechASRConfig(name="clean", description="'Clean' speech."),
+        LibrispeechASRConfig(name="other", description="'Other', more challenging, speech."),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "file": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "speaker_id": datasets.Value("int64"),
+                    "chapter_id": datasets.Value("int64"),
+                    "id": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=("speech", "text"),
+            homepage=_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download_and_extract(_DL_URLS[self.config.name])
+
+        if self.config.name == "clean":
+            train_splits = [
+                datasets.SplitGenerator(name="train.100", gen_kwargs={"archive_path": archive_path["train.100"]}),
+                datasets.SplitGenerator(name="train.360", gen_kwargs={"archive_path": archive_path["train.360"]}),
```

Comment:
What's the difference between train.100 and train.360 ?

Reply:
From the paper:

> The size of the corpus makes it impractical, or at least inconvenient
> for some users, to distribute it as a single large archive. Thus the
> training portion of the corpus is split into three subsets, with approximate size 100, 360 and 500 hours respectively. A simple automatic
> procedure was used to select the audio in the first two sets to be, on
> average, of higher recording quality and with accents closer to US
> English. An acoustic model was trained on WSJ’s si-84 data subset
> and was used to recognize the audio in the corpus, using a bigram
> LM estimated on the text of the respective books. We computed the
> Word Error Rate (WER) of this automatic transcript relative to our
> reference transcripts obtained from the book texts
> ...

> The rest of the audio in the “clean” pool was randomly split into
> two training sets with approximate size 100 and 360 hours respectively. For each speaker in these training sets the amount of speech
> was limited to 25 minutes, in order to avoid major imbalances in
> per-speaker audio duration

> ...

> The “other” pool was similarly split into test and development
> sets, and a single training set of approximately 500 hours.

I'll update the README.md to make this clearer